### ISSUE TITLE
【Reopen】What target would you like to use?

### ISSUE THREAD

**Issue by josercc** [fastlane/fastlane#12177]
It seems that the problem has not yet been completely resolved.
When there are two targets in my project, for example creating a project will automatically bring the two targets of test and UI test so there are three targets.
This will prompt me to choose the target, it blocks the automation of the script.
When I delete only one, it will not be prompted again.
The target we choose to package is set in the gym method

https://github.com/fastlane/fastlane/issues/12115

**Comment by Tayphoon** [comment]
Same here

**Comment by joshdholtz** [comment]
@josercc @Tayphoon Is this happening to you both on `get_version_number`?

**Comment by pokono** [comment]
+1

**Comment by amitjpatel** [comment]
+1

**Comment by josercc** [comment]
@joshdholtz 

![](http://ipicimage-1251019290.coscd.myqcloud.com/2018-03-28-002652.jpg)

**Comment by pokono** [comment]
Yes! I solved my issue specifying the `target` on the `get_version_number` action.

**Comment by joshdholtz** [comment]
Hey everyone! The solution to this is to specify a target in the options with `target: "your-target-name"`. We had to make an internal change to how `get_latest_version` worked. 

## History time (it worked but not correctly)
`get_lastest_verson` used to use `agvtool` under the covers to get the versions of all the targets. This required that weird setup In "Build Settings" (which seemed a little unnecessary). Also, it wasn't actually getting versions of "targets" but actually getting versions of `Info.plist`'s (which is where things get confusing). We would then take that CLI response from `agvtool` and try to find the version value for your target.

^ Here is where it got dangerous

`agvtool` didn't actually give us a target name with the version. It gave us an `Info.plist` path (for some target) with a version. So we were assuming that your target and `Info.plist` were named the same (which they might not be).

## Time for a change (more stability and easier to support)
This is why we had to make the change. We were "looking for a target" but basing it off of an `Info.plist` name that may or may not match your target **while** also requiring you to do adjust "Build Settings" to get working.

Our change that we did in `2.87.0` was to get rid of `agvtool` completely (along with the "Build Settings" requirement) and instead use `xcodeproj` gem that could **actually** get a target and its version. We added some logic in there in `2.87.1` and `2.87.2` to attempt to pick a target (if not specified) if it is the only target (and ignore test targets). This would solve some issues of not specifying a target but not all (which is where we stand today on this issue).

## TL;DR
The previous implementation with `agvtool` would "get lucky" most times when trying to find the person but could also get "unlucky" (which the "unlucky" being really nasty to debug and causing more issues). The previous implementation was unstable and required answering multiple Github issues explaining why it behaved the way it did.

The new implementation is more stable and actually grabs the correct target and will be better for _fastlane_ going forward but **may require some previous implementation** to specify a target with `target: "your-target-name"`

Hope that does it :blush:
